addons/purchase_requisition/models/purchase.py

```diff
@@ -212,34 +212,46 @@ def action_compare_alternative_lines(self):
         }
 
     def get_tender_best_lines(self):
-        product_to_best_price_line = defaultdict(lambda: self.env['purchase.order.line'])
-        product_to_best_date_line = defaultdict(lambda: self.env['purchase.order.line'])
-        product_to_best_price_unit = defaultdict(lambda: self.env['purchase.order.line'])
+        product_to_best_price_line = defaultdict(lambda: [self.env['purchase.order.line'], False])
+        product_to_best_date_line = defaultdict(lambda: [self.env['purchase.order.line'], False])
+        product_to_best_price_unit = defaultdict(lambda: [self.env['purchase.order.line'], False])
```

Comment:
the False contain the best price_subtotal/date/price_unit for the given product, the [first version](https://github.com/odoo/odoo/commit/5669138347dc209599fb3bb47a1d7716f012da6e) of this PR used a dictionary instead (and had ['lines'] and ['best_*'] instead of [0] and [1].

with your commit about using currency.round() and keeping the code, it would not be necessary

Reply:
Ah I see now. Indeed we do need to make sure that when we compare the stored best values in the correct currency. Instead of making an extra complicated data structure that makes the code harder to read, can we just check if there are more than 1 currencies used in the POs (i.e. check if a currency_rates variable is set from the pre-calculation I suggested earlier) and reconvert it if there are + the (first) current best line's currency isn't the company's currency?

Also, now that I understand the reasoning for the change, there's no need to change the `product_to_best_date_line` structure to be more complex only to match the structure of the monetary values since it only makes it more confusing (i.e. it doesn't make sense to store the best date when we're never going to use it)